platforms/ide/problems-rendering/src/main/java/org/gradle/problems/internal/rendering/ProblemRenderer.java

```diff
@@ -41,25 +37,16 @@ public ProblemRenderer(Writer writer) {
     }
 
     public void render(List<Problem> problems) {
-        Map<ProblemId, List<Problem>> renderingGroups = new HashMap<>();
-        for (Problem problem : problems) {
-            List<Problem> groupedProblems = renderingGroups.computeIfAbsent(
-                problem.getDefinition().getId(),
-                id -> new ArrayList<>()
-            );
-            groupedProblems.add(problem);
-        }
-
-        renderingGroups.forEach((id, groupedProblems) -> renderProblemGroup(output, id, groupedProblems));
```

Comment:
This was the problem. We only added the newlines between rendered reports within a group, but not between groups. 

Grouping here is not a desired behaviour, at least not at this point.It's producer's responsibility to define, in which order the problems should be emitted and displayed. 

Later, when we have more integration, we can come back and see how we can improve the rendered data, but rn it's just premature optimization.

Reply:
Even though I don't think it was premature, I accept the choice here.